### ISSUE TITLE
Null Default doesn't follow the specs

### ISSUE THREAD

**Issue by bka9** [klarna/erlavro#66]
```
 {
     "namespace": "com.example",
      "type": "record",
      "name": "Record",
       "fields": [
           {"name": "UUID", "type": ["null", "string"], "default": "null"},
           {"name": "Version", "type": ["null", "string"], "default": "null"}
        ]
}
```
The above is valid avro specification. But when the make_simple_decoder is applied to it, it fails complaining about a bad default. If you remove the quotes from the null it works. It should also match the string "null" when determining avro null types.

**Comment by zmstone** [comment]
I don't think this is a valid avro schema.
`"null"` type's JSON encoding result is `null` without quotes.
A quoted null is "string" type.
A workaround is to call `avro:decode_schema/2` with `ignore_bad_default_values` option set to `true`.

However, according to avro specification:
> (Note that when a default value is specified for a record field whose type is a union, the type of the default value must match the first element of the union. Thus, for unions containing "null", the "null" is usually listed first, since the default value of such unions is typically null.)

Meaning the default value given in schema in this case has to be `null` regardless quoted or not.

**Comment by bka9** [comment]
> For example, the union schema ["null","string","Foo"], where Foo is a record name, would encode:
>
> null as null;
> the string "a" as {"string": "a"}; and
> a Foo instance as {"Foo": {...}}, where {...} indicates the JSON encoding of a Foo instance.

**Comment by bka9** [comment]
Okay, I am convinced now, funny though in every other avro library in other languages and I have used (java, go, c#) their schema parsers have not complained. That is what threw me for a loop.

**Comment by zmstone** [comment]
fix released as 2.6.1
